include/seqan3/range/view/take.hpp

```diff
@@ -362,23 +372,32 @@ class view_take : public ranges::view_interface<view_take<urng_t, exactly, or_th
      *
      * No-throw guarantee.
      */
-    constexpr iterator begin() noexcept
+    constexpr auto begin() noexcept
     {
-        return {seqan3::begin(urange), 0, target_size, &(*this)};
+        if constexpr (std::ranges::RandomAccessRange<urng_t> && std::ranges::SizedRange<urng_t>)
+            return std::ranges::begin(urange);
+        else
+            return iterator{std::ranges::begin(urange), 0, target_size, this};
     }
 
     //!\copydoc begin()
-    constexpr const_iterator begin() const noexcept
+    constexpr auto begin() const noexcept
         requires ConstIterableRange<urng_t>
     {
-        return {seqan3::cbegin(urange), 0, target_size};
+        if constexpr (std::ranges::RandomAccessRange<urng_t> && std::ranges::SizedRange<urng_t>)
+            return std::ranges::cbegin(urange);
+        else
+            return const_iterator{std::ranges::cbegin(urange), 0, target_size};
```

Comment:
why does this iterator does not need this? this is on purpose?

Reply:
Only single-pass iterators need to pass a pointer/reference to self, because in that case iterating consumes the range and the size of the take becomes smaller with every step.

The const version of the iterator doesn't exist for pure input ranges so it's not neccessary.

(For ForwardRanges the parameter is ignore)

Reply:
Ok